src/condition.cpp

```diff
@@ -780,7 +780,7 @@ void conditional_t<T>::set_has_stolen_item( bool is_npc )
         bool found_in_inv = false;
         for( auto &elem : actor->inv_dump() ) {
             if( elem->has_old_owner() ) {
-                if( elem->get_old_owner() == p.get_faction()->id ) {
+                if( elem->is_old_owner( p ) ) {
```

Comment:
you should collapse `has_old_owner()` and `is_old_owner()` into a single function.

Reply:
something like .
`bool item::is_available_to_take( Character &c )
{
    return owner.is_null() || is_owned_by( c );
}`  ?